src/mount_efs/__init__.py

```diff
@@ -456,6 +457,7 @@ def get_aws_security_credentials_from_webidentity(role_arn, token_file, is_fatal
         else:
             return None, None
 
+    STS_ENDPOINT_URL = STS_ENDPOINT_URL_FORMAT.format(region)
```

Comment:
Instead of making region as a parameter

Reply:
@Cappuccinuo 
I want the default behavior, not the settings. This is because running in a region far from the Global Services data center results in a timeout error. It is nonsense to have them set to avoid it. So I want to use the region endpoint by default.

If getting the region is a problem outside of webidentity, can I fix it to get the region within get_aws_security_credentials_from_webidentity?

Reply:
@Cappuccinuo 
If the above policy is not good, can I extend the timeout? Since it takes less than 1 second from the Tokyo region to the global service endpoint (2 seconds is enough), can I change it to 5 seconds?
https://github.com/aws/efs-utils/blob/master/src/mount_efs/__init__.py#L561

Reply:
In fact, I think it will be a good choice to always talk to regional sts endpoint to reduce latency, increase the timeout will also influence the other url request, we don't want to give too long for the url request timeout. 

Thus I would recommend when retrieve the credential for web identity, use regional endpoint always and store the region/endpoint url info in the state file so that watchdog can refresh the credentials accordingly.

Reply:
@Cappuccinuo 
Thanks you. So I will fix it to get the region in get_aws_security_credentials_from_webidentity method, store it in the state file and reuse it.

Reply:
Sounds good, thanks!

Reply:
@Cappuccinuo 
I'm sorry, I reviewed the source again, but after all, I came to the conclusion that there is no problem with the current fixes.

1. You pointed out that it's not good to pass a region as a parameter from the root, but when I read the code again, it looks like I'm passing the region parameter in multiple places in the existing code. Also, if I fix it so that it doesn't pass region as a parameter, it doesn't seem to make much sense because it will be necessary to pass the config object as a parameter from the root this time.

2. The matter is to store the region in a state file so that watchdog can reuse it, but this has already been realized in the existing code. So it didn't seem like any additional modifications were needed. On the watchdog implementation side, the region obtained from the state file is passed as a parameter. This time, I just modified the region parameter to pass to the get_aws_security_credentials_from_webidentity method.
https://github.com/aws/efs-utils/blob/cd27177d80d7a3a5f44bc3412c0b275c6ba3eece/src/watchdog/__init__.py#L676

Reply:
Seems reasonable to me, I thought my original attempt is finding a way that can avoid passing region around, but the config is necessary to get the region.

Can you make the commit message more descriptive and remove all the `Fix` sentences? Thanks.

Reply:
@Cappuccinuo 
Thank you for confirmation. I fixed the commit message.